sherpa/plot/pylab_backend.py

```diff
@@ -19,857 +19,822 @@
-           'get_fit_contour_defaults', 'get_resid_contour_defaults',
-           'get_ratio_contour_defaults', 'get_confid_plot_defaults',
-           'get_confid_contour_defaults', 'set_window_redraw', 'set_jointplot',
-           'get_model_histo_defaults', 'get_histo_defaults',
-           'get_component_plot_defaults', 'get_component_histo_defaults',
-           'vline', 'hline', 'get_scatter_plot_defaults',
-           'get_cdf_plot_defaults', 'get_latex_for_string', 'name',
-           'initialize_plot', 'select_plot')
-
-
-# Name of the backend
-name = 'pylab'
+__all__ = ['PylabBackend']
 
 logger = logging.getLogger(__name__)
 
+# Most of the kwargs docstrings are still true, but some can be updated
+# because matplotlib does support a wider range than just the
+# set of Sherpa backend-independent options.
+updated_kwarg_docs = {
+    'color': ['string or tuple', 'Any matplotlib color'],
+    'linecolor': ['string or tuple', 'Any matplotlib color'],
+    'marker': ['string',
+               '''"None" (as a string, no marker shown), "" (empty string, no marker shown),
+or any matplotlib marker, e.g. any of `os+v` or others
+(see matplotlib documentation).'''],
+    'linestyle': ['string',
+                  '''``'noline'``,
+``'None'`` (as string, same as ``'noline'``),
+``'solid'``, ``'dot'``, ``'dash'``, ``'dotdash'``, ``'-'`` (solid
+line), ``':'`` (dotted), ``'--'`` (dashed), ``'-.'`` (dot-dashed),
+``''`` (empty string, no line shown), `None` (default - usually
+solid line) or any other matplotlib linestyle.'''],
+    'drawstyle': ['string', 'matplolib drawstyle'],
+}
 
-def begin():
-    pass
-
-
-def end():
-    set_window_redraw(True)
-    if plt.isinteractive():
-        plt.draw()
-
-
-def exceptions():
-    pass
-
-
-def clear_window():
-    plt.clf()
-
-
-def set_window_redraw(redraw):
-    if redraw:
-        plt.draw()
-
-
-def setup_axes(overplot, clearwindow):
-    """Return the axes object, creating it if necessary.
-
-    Parameters
-    ----------
-    overplot : bool
-    clearwindow : bool
-
-    Returns
-    -------
-    axis
-        The matplotlib axes object.
-    """
-
-    # Do we need to clear the window?
-    if not overplot and clearwindow:
-        clear_window()
-
-    return plt.gca()
-
-
-def setup_plot(axes, title, xlabel, ylabel, xlog=False, ylog=False):
-    """Basic plot setup.
-
-    Parameters
-    ----------
-    axes
-        The plot axes (output of setup_axes).
-    title, xlabel, ylabel : str or None
-        The plot, x-axis, and y-axis titles. They are skipped if
-        the empty string or None.
-    xlog , ylog : bool
-        Should the scale be logarithmic (True) or linear (False)?
-
-    """
-
-    axes.set_xscale('log' if xlog else 'linear')
-    axes.set_yscale('log' if ylog else 'linear')
-
-    if title:
+kwargs_doc = ChainMap(updated_kwarg_docs, orig_kwargs_doc)
+
+
+class PylabBackend(BasicBackend):
+"""sherpa plotting backend using matplotlib"""
```

Comment:
```suggestion
    """sherpa plotting backend using matplotlib"""
```

stupid GH suggestion logic...